src/Dto/Segments/CreateSegment.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Dto\Segments;
+
+use OneSignal\Dto\AbstractDto;
+use OneSignal\Dto\Filters\AbstractFilter;
+
+class CreateSegment implements AbstractDto
+{
+    /**
+     * @var non-empty-string
+     */
+    protected string $name;
+
+    /**
+     * @var array<int, AbstractFilter>
+     */
+    protected array $filters = [];
+
+    /**
+     * @param non-empty-string           $name
+     * @param array<int, AbstractFilter> $filters
+     */
+    public function __construct(string $name, array $filters = [])
+    {
+        $this->name = $name;
+        $this->filters = $filters;
+    }
```

Comment:
As these are just DTO's, we should let them construct without any requirements before and let the user decide what to fill and what not.

```suggestion
    /**
     * @param non-empty-string|null     $name
     * @param list<AbstractFilter>|null $filters
     */
    public function __construct(string $name = null, array $filters = null)
    {
        $this->name = $name;
        $this->filters = $filters;
    }
```